api/base/filters.py

```diff
@@ -61,6 +66,83 @@ def filter_queryset(self, request, queryset, view):
         return queryset
 
 
+class NewDjangoFilterMixin(django_filters.FilterSet):
+
+    QUERY_PATTERN = re.compile(r'^filter\[(?P<fields>((?:,*\s*\w+)*))\](\[(?P<op>\w+)\])?$')
+    FILTER_FIELDS = re.compile(r'(?:,*\s*(\w+)+)')
+
+    or_fields = []
+
+    def __init__(self, data=None, *args, **kwargs):
+        if data:
+            new_qd = QueryDict('', mutable=True)
+            for key, value in data.iteritems():
+                match = self.QUERY_PATTERN.match(key)
+                if match:
+                    match_dict = match.groupdict()
+                    fields = match_dict['fields']
+                    field_names = re.findall(self.FILTER_FIELDS, fields.strip())
+
+                    self.or_fields = []
+                    multiple_values = value.split(',')
+                    if len(multiple_values) > 1 or len(field_names) > 1:
```

Comment:
the len of `multiple_values` is checked here cause if there are more than one, those also need to be sent to the or_fields list for a built `OR` query, independent if there are multiple field names